junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/ScriptExecutionCondition.java

```diff
@@ -62,115 +64,116 @@ public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext con
+		 * {@code java.scripting} module.
+		 *
+		 * @see Class#forName(String)
+		 */
+		static Evaluator forName(String name) {
+			// Assert that "javax.script.ScriptEngine" is loadable via basic reflection.
+			return forName("javax.script.ScriptEngine", name);
 		}
-		catch (ScriptException e) {
-			throw new ScriptEvaluationException("Script evaluation failed for: " + script.getAnnotationAsString(), e);
-		}
-	}
 
-	ConditionEvaluationResult computeConditionEvaluationResult(Script script, Object result) {
-		// Treat "null" result as an error.
-		if (result == null) {
-			throw new ScriptEvaluationException("Script returned `null`: " + script.getAnnotationAsString());
+		static Evaluator forName(String nameOfScriptEngine, String name) {
+			// Assert that precondition name is loadable via basic reflection.
+			try {
+				Class.forName(nameOfScriptEngine);
+			}
+			catch (Throwable cause) {
+				String message = "Class `" + nameOfScriptEngine + "` is not loadable, " //
+						+ "script-based test execution is disabled. " //
+						+ "If the originating cause is a `NoClassDefFoundError: javax/script/...` and " //
+						+ "the underlying runtime environment is executed with an activated module system " //
+						+ "(aka Jigsaw or JPMS) you need to add the `java.scripting` module to the " //
+						+ "root modules via `--add-modules ...,java.scripting`";
+				return new ThrowingEvaluator(message, cause);
+			}
+			// Now create the evaluator instance specified by its class name.
+			try {
+				return (Evaluator) Class.forName(name).getDeclaredConstructor().newInstance();
+			}
+			catch (ReflectiveOperationException cause) {
+				String message = "Creating instance of class `" + name + "` failed," //
+						+ "script-based test execution is disabled.";
+				return new ThrowingEvaluator(message, cause);
+			}
 		}
 
-		// Trivial case: script returned a custom ConditionEvaluationResult instance.
-		if (result instanceof ConditionEvaluationResult) {
-			return (ConditionEvaluationResult) result;
-		}
+	}
 
-		String resultAsString = String.valueOf(result);
-		String reason = script.toReasonString(resultAsString);
+	/**
+	 * Evaluator implementation that always throws an {@link ScriptEvaluationException}.
+	 */
+	static class ThrowingEvaluator implements Evaluator {
 
-		// Cast or parse result to a boolean value.
-		boolean isTrue;
-		if (result instanceof Boolean) {
-			isTrue = (Boolean) result;
-		}
-		else {
-			isTrue = Boolean.parseBoolean(resultAsString);
-		}
+		final ScriptEvaluationException exception;
 
-		// Flip enabled/disabled result based on the associated annotation type.
-		if (script.getAnnotationType() == EnabledIf.class) {
-			return isTrue ? enabled(reason) : disabled(reason);
-		}
-		if (script.getAnnotationType() == DisabledIf.class) {
-			return isTrue ? disabled(reason) : enabled(reason);
+		ThrowingEvaluator(String message, Throwable cause) {
+			this.exception = new ScriptEvaluationException(message, cause);
```

Comment:
Pondering the idea to introduce a new `JUnitException` specialization here. Something like `ScriptingConfigurationException`. As there is factually no "evaluation" going on.

Reply:
Hmmmm... if evaluation is not the cause, maybe it's best just to throw an `ExtensionConfigurationException` and get rid of the `ScriptEvaluationException` subclass (if it then becomes unused).

Reply:
Or are you saying you want `ScriptingConfigurationException` to replace `ScriptEvaluationException`?

Reply:
Going to use `ExtensionConfigurationException` in the `ThrowingEvaluator`. The `ScriptEvaluationException` is still used for when a user-defined script contains (syntax) errors.